### ISSUE TITLE
Question: Way to output the found string obfuscation techniques

### ISSUE THREAD

**Issue by Homette** [CalebFenton/simplify#114]
Hi there, 
I am using this tool to decrypt strings in apks and I'd like to know if there is a way to output the string obfuscation techniques that are being used?
Regards

**Comment by CalebFenton** [comment]
Interesting question. I've never thought about it.

It really depends on your taxonomy of string obfuscation techniques. If your taxonomy is "uses xor" and "not uses xor" then yeah, it's possible. If you want classifications that are a less trivial, it gets more difficult and prone to being probabilistic, I reckon.

What are some techniques you're thinking of? When I think string encryption, I broadly classify into something like:

* base 64
* xor encryption
* cipher shift
* two-way encryption (AES, RC4, etc.)
* custom encoder (catch-all for everything else)
* java / native (as a modifier)

**Comment by Homette** [comment]
I was thinking about the techniques you've listed.

**Comment by CalebFenton** [comment]
Why? What problem are you trying to solve?

**Comment by Homette** [comment]
I am working on my master thesis and I am statically analysing Android malware. I stumbled across your tool while figuring out how to extract (obfuscated) strings from the APKs. I am using your tool to decrypt strings that are afterwards extracted. Then I thought that it would be nice to know which obfuscation techniques were used and put that in an extra subchapter. 
That's why I wanted to know if it's possible to output the techniques.

**Comment by CalebFenton** [comment]
I would estimate that it's much, much easier to simply look at the code and find xor operations, base64 api calls, encryption api calls, etc. Manual analysis shouldn't be difficult. This type of analysis is more suited to a project like APKiD. It's not really within the scope of deobfuscation, so I'll close the issue. Feel free to continue discussion.

**Comment by Homette** [comment]
Yes, it would be easier, but I have about 27000 files and examining every single file manually is just not possible that's why I wanted to automatise it and hoped that it would be more or less quite easy to adjust this tool to get the information.

**Comment by CalebFenton** [comment]
Oh I totally get wanting to automate it. I think yara and regex could work pretty well. Smalivm would be probablistic at best, and highly unstable at worst.  Sorry bro!
